compiler/rustc_target/src/abi/call/mod.rs

```diff
@@ -517,6 +522,11 @@ impl<'a, Ty> ArgAbi<'a, Ty> {
         match self.mode {
             PassMode::Indirect { ref mut attrs, extra_attrs: _, ref mut on_stack } => {
                 *on_stack = true;
+                // If the argument is passed by value, the `NoAlias` attribute should never be
+                // applied. More information about this in:
+                //  * https://lists.llvm.org/pipermail/llvm-commits/Week-of-Mon-20171218/512066.html
+                //  * https://reviews.llvm.org/D40118
+                attrs.unset(ArgAttribute::NoAlias);
```

Comment:
The linked patch is unrelated to this change. The patch allows _more_ `noalias` when an argument aliases with a `byval` argument, whereas this change is removing `noalias`. (Also that patch is about `noalias` applied to non-`byval` arguments, whereas this change is for `noalias` applied to `byval` arguments...)

Reply:
Ah indeed, it links to callsite, not function args.

Reply:
That's not the distinction. `noalias` has the same meaning whether it's applied on the callsite's argument or the function declaration's argument, and the code you're changing affects both.

To put it more simply:
- that patch is saying "noalias is okay in more places"
- your change, by removing `noalias`, is saying "noalias is okay in fewer places"

Now, this isn't a direct contradiction, since they're _also_ talking about different situations. 🙃 It's just unrelated.